src/popup/Popup.tsx

```diff
@@ -0,0 +1,202 @@
+import { fetchCsv, metchCsv } from "./downloadCsv";
+import classnames from "classnames";
+
+const TABS = {
+    search: "search",
+    settings: "settings",
+} as const;
+
+export const PopupPage: React.FC<{}> = () => {
+    const iconStyle = { width: rem(12), height: rem(12) };
+    const [activeTab, setActiveTab] = React.useState<string>(TABS.search);
+    return (
+        <Tabs className="main-page" value={activeTab} onChange={val => val != null && setActiveTab(val)} color="indigo">
+
+            <Tabs.List>
+                <Tabs.Tab value={TABS.search}>Search</Tabs.Tab>
+                <Tabs.Tab value={TABS.settings} leftSection={<IconSettings style={iconStyle} />}>Settings</Tabs.Tab>
+            </Tabs.List>
+
+            <Tabs.Panel value={TABS.search} className={classnames("tab-panel", "search-panel")}>
+                <SearchPage />
+            </Tabs.Panel>
+
+            <Tabs.Panel value={TABS.settings} className={"tab-panel"}>
+                <SettingsPage />
+            </Tabs.Panel>
+
+        </Tabs>
+    );
+}
+
+const SettingsPage: React.FC<{}> = () => {
+    return (
+        <>
+            <Text c="dimmed" size="sm">
+                We can populate this nicely with{" "}
+                <Text span inherit c="var(--mantine-color-anchor)">
+                    links to stuff
+                </Text>{" "}
+                and some settings. Right now we only have one.
+            </Text>
+            <EnabledSwitch />
+        </>
+    );
+}
+
+function EnabledSwitch() {
+    const [checked, setChecked] = React.useState(false);
+    const fetchIsEnabled = React.useCallback(async () => {
+        try {
+            const isEnabled = await isExtensionEnabled();
+            setChecked(isEnabled ?? false);
+        } catch (e) {
+            console.error("Could not read `isEnabled`", e);
+        }
+    }, []);
+
+    React.useEffect(() => { fetchIsEnabled(); }, [fetchIsEnabled]);
+
+    const toggleEnabled = React.useCallback(async () => {
+        await setIsExtensionEnabled(!checked);
+        fetchIsEnabled();
+    }, [checked, fetchIsEnabled]);
+
+    return (
+        <Switch
+            checked={checked}
+            onChange={toggleEnabled}
+            label={"Enabled"}
+        />
+    );
+}
+
+const SearchPage: React.FC<{}> = () => {
+    const [companyNames, setcompanyNames] = React.useState<"loading" | "error" | string[]>("loading");
+    const fetchCompanies = React.useCallback(async () => {
+        try {
+            const companyData = await getCompanyData();
+            setcompanyNames(companyData.companies);
+        } catch (e) {
+            console.error("Could not read `company data`", e);
+            // setcompanyNames([]);
+            setcompanyNames("error");
+        }
+    }, []);
+
+    React.useEffect(() => { fetchCompanies(); }, [fetchCompanies]);
+
+    if (companyNames == "loading") {
+        return <Loader />
+    }
+
+    // TODO: maybe expose means to handle errors here
```

Comment:
We could perhaps assume that the data isn't there, go down the "download and save" code path, and attempt to re-write?